pkg/okteto/auth.go

```diff
@@ -140,6 +142,26 @@ func saveAuthData(user *User, url string) error {
 }
 
 func queryUser(ctx context.Context, client *graphql.Client, token string) (*q, error) {
+	var user q
+	q := `query {
+		user {
+			id,name,email,externalID,token,new,registry,buildkit,certificate,globalNamespace
+		}}`
+
+	req := getRequest(q, token)
+
+	if err := client.Run(ctx, req, &user); err != nil {
+		if strings.Contains(err.Error(), "Cannot query field \"globalNamespace\" on type \"me\"") {
```

Comment:
Maybe it is out of the scope of this PR, but maybe we can start using IntrospectionQueries to this kind of things: https://graphql.org/learn/introspection/.

Executing this query against my dev cluster

```
query IntrospectionQuery {
  __type(name: "me"){
    name
    fields {
      name
    }
  }
}
``` 

I got

```
{
  "data": {
    "__type": {
      "fields": [
        {
          "name": "avatar"
        },
        {
          "name": "buildkit"
        },
        {
          "name": "capabilities"
        },
        {
          "name": "certificate"
        },
        {
          "name": "config"
        },
        {
          "name": "email"
        },
        {
          "name": "externalID"
        },
        {
          "name": "gitSSHSupport"
        },
        {
          "name": "githubID"
        },
        {
          "name": "id"
        },
        {
          "name": "integrations"
        },
        {
          "name": "name"
        },
        {
          "name": "namespace"
        },
        {
          "name": "new"
        },
        {
          "name": "onboarded"
        },
        {
          "name": "personalAccessTokens"
        },
        {
          "name": "plan"
        },
        {
          "name": "planSubscribed"
        },
        {
          "name": "privateEndpointToken"
        },
        {
          "name": "quickstarts"
        },
        {
          "name": "quotaPlan"
        },
        {
          "name": "registry"
        },
        {
          "name": "secrets"
        },
        {
          "name": "super"
        },
        {
          "name": "token"
        },
        {
          "name": "trialExpiration"
        },
        {
          "name": "userDefinedNamespaces"
        }
      ],
      "name": "me"
    }
  }
}
```

I think this is most robust since it is based on the graphql schema defined on the server

Reply:
Should we open a ticket for this?

Reply:
Yeap, I would implement it right after this is merge

Reply:
Yes, we can merge this PR as it is now and include the query instrospection later if we see the value it could provide, but do not block this PR for that